op/model.py

```diff
@@ -48,31 +49,58 @@ def get_unit(self, unit_name):
 
 
 class ModelCache(weakref.WeakValueDictionary):
+
+    def __init__(self, *args, **kw):
+        self.backend = kw.pop('backend')
+        super().__init__(*args, **kw)
```

Comment:
We can simplify this logic by dropping the base class, which we're not really using I believe, and having this instead:
```
class ModelCache:
    def __init__(self, backend):
        self._backend = backend
        self._weakrefs = weakref.WeakValueDictionary()
```

Then, the prior line constructing it can be the more straightforward:
```
self._cache = ModelCache(backend)
```
No kwargs.